packages/sdk/connect/src/connection.ts

```diff
@@ -341,20 +326,27 @@ export class Connection {
   sendSignedTransaction = async (signedTransactionData: string): Promise<TransactionResult> => {
     return toTxResult(this.web3.eth.sendSignedTransaction(signedTransactionData))
   }
+  // if neither gas price nor feeMarket fields are present set them.
+  setFeeMarketGas = async (tx: CeloTx): Promise<CeloTx> => {
+    // default to the current values
+    const calls = [Promise.resolve(tx.maxFeePerGas), Promise.resolve(tx.maxPriorityFeePerGas)]
 
-  /** @deprecated no longer needed since gasPrice is available on node rpc */
-  fillGasPrice(tx: CeloTx): CeloTx {
-    if (tx.feeCurrency && tx.gasPrice === '0' && this.currencyGasPrice.has(tx.feeCurrency)) {
-      return {
-        ...tx,
-        gasPrice: this.currencyGasPrice.get(tx.feeCurrency),
+    if ((isEmpty(tx.gasPrice) && isEmpty(tx.maxFeePerGas)) || isEmpty(tx.maxPriorityFeePerGas)) {
+      if (isEmpty(tx.maxFeePerGas)) {
+        calls[0] = this.gasPrice(tx.feeCurrency)
+      }
```

Comment:
this if statements seem interesting, could we make simpler, looks like we are checking tx.maxFeePerGas multiple times?